### PR TITLE
Disabled bsync function by default, added function to enable it.

### PR DESCRIPTION
**Pull request by andrewbenton**
By default, barrier sync functionality is disabled by default with the PR.  By calling setCanBarrier(), the barrier sync can be re-enabled on a test-wise level.  In the case that the barrier sync is not enabled for a test, and one of that test's glue functions calls Barrier.sync(), Zucchini will throw a ZucchiniRuntimeException.

### CODE REVIEW

File: src/main/java/com/comcast/zucchini/AbstractZucchiniTest.java [hunk 1]
```diff
@@ -132,6 +147,8 @@ private boolean envSerialized() {
      * @param contexts The list of contexts that will be run.
      */
     public void runParallel(List<TestContext> contexts) {
+        this.canBarrierInTest = this.canBarrier;
```
Comment (dimaj) [inline src/main/java/com/comcast/zucchini/AbstractZucchiniTest.java]: instead of setting this property from both runParallel and runSerial, why not call it once from run?

Comment (andrewbenton) [inline src/main/java/com/comcast/zucchini/AbstractZucchiniTest.java]: With the API, runParallel() and runSerial() are both public methods.  Since they are public, the run() method is not always called.  In the case that someone decided to call the bare runParallel() and runSerial() methods, this method prevents bad behavior on the part of zucchini.  We _could_ put something in the docs about undefined behavior on using a barrier and calling the bare runParallel() and runSerial() methods, and then we could put the statement just in the run() method, but I think that's less stable.  

tl;dr - The current method of doing this is the belt-and-suspenders approach.  For the next api change, it might desirable to make the runParallel() and runSerial() private, and provide an overloaded run() method so that it defaults in accordance with envSerialized(), and can be overridden with run(boolean).
